### PR TITLE
Bugfix/acquires must be combined in assemble

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Although we've begun our migration to single channel AcquisitionInstructions, the Qobj does not support this style. This is necessary to combine acquires by time.

Also have to remove deprecations that are still being used in assemble. (for warnings)

### CODE REVIEW

File: qiskit/assembler/assemble_schedules.py [hunk 1]
```diff
@@ -109,18 +107,34 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
                         channel=instruction.channels[0])
                 # add samples to pulse library
                 user_pulselib[name] = instruction.command
-            elif isinstance(instruction, AcquireInstruction):
+
+            if isinstance(instruction, AcquireInstruction):
                 max_memory_slot = max(max_memory_slot,
                                       *[slot.index for slot in instruction.mem_slots])
+                acquire_instruction_map[(shift, instruction.command)].append(instruction)
```
Comment (taalexander) [inline qiskit/assembler/assemble_schedules.py]: I think a comment here about why this is needed would probably be pretty helpful in the future.

File: qiskit/assembler/assemble_schedules.py [hunk 2]
```diff
@@ -109,18 +107,34 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
...
 
-        if meas_map:
-            # verify all acquires satisfy meas_map
-            _validate_meas_map(acquire_instructions, meas_map)
+            qobj_instructions.append(instruction_converter(shift, instruction))
+
+        if acquire_instruction_map:
+            if meas_map:
+                _validate_meas_map(acquire_instruction_map, meas_map)
+            for (shift, _), instructions in acquire_instruction_map.items():
```
Comment (taalexander) [inline qiskit/assembler/assemble_schedules.py]: Could this be broken into a function like `_validate_meas_map`?

Comment (lcapelluto) [reply qiskit/assembler/assemble_schedules.py]: Sure

Comment (lcapelluto) [reply qiskit/assembler/assemble_schedules.py]: also think I will refactor this file to make a few separate functions (e.g. assemble_instructions)

Comment (lcapelluto) [reply qiskit/assembler/assemble_schedules.py]: not in this PR though

**Comment by lcapelluto** [comment]
Job successfully ran on a device, using a schedule w single channel acquires. (MattT is looking into the build failures now)

**Pull request merged**